.github/workflows/build.yml

```diff
@@ -0,0 +1,61 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.17.x
+
+    - name: check assets
+      run: |
+        make assets
+        echo "Checking if 'make assets' was ran after adding files..."
+        git diff --exit-code
+
+    - name: check dependencies
+      run: |
+        go mod tidy -v
+        if [[ -n $(git --no-pager diff --exit-code go.mod go.sum) ]]; then echo "There are unused dependencies that should be removed. Please execute `go mod tidy` to fix it."; exit 1; fi
+
+    - name: Run build
+      run: make build
+
+  image:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: Build and push
+      uses: docker/build-push-action@v2
+      with:
+        file: ./Dockerfile
+        push: false
+        tags: not-to-push/mattermod:latest
```

Comment:
why do you need the `not-to-push` as you have the `push: false`? Also why not just typical `make build-image`. I am a bit confused here

Reply:
this is just to check if the docker build works we are not pushing images for prs

Reply:
updated the name
thanks for the review